### ISSUE TITLE
TensorFlow 2.0 support

### ISSUE THREAD

**Issue by yongtang** [tensorflow/io#55]
This issue is created to track TensorFlow 2.0 support for tensorflow-io. The most visible changes in TensorFlow 2.0 is the eager execution in default. Will need to test thoroughly and see if there are any impact on tensorflow-io.

**Comment by yongtang** [comment]
When trying tensorflow-io's build with TF 1.13, I noticed that the generated binary for 1.13 may not be compatible with TF 2.0. Here is some core dump:
```
(gdb) where
#0  __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:51
#1  0x00007fd30249c801 in __GI_abort () at abort.c:79
#2  0x00007fd3024e5897 in __libc_message (action=action@entry=do_abort, fmt=fmt@entry=0x7fd302612b9a "%s\n") at ../sysdeps/posix/libc_fatal.c:181
#3  0x00007fd3024ec90a in malloc_printerr (str=str@entry=0x7fd302610d88 "free(): invalid pointer") at malloc.c:5350
#4  0x00007fd3024f3e1c in _int_free (have_lock=0, p=0x25f6798, av=0x7fd302847c40 <main_arena>) at malloc.c:4157
#5  __GI___libc_free (mem=0x25f67a8) at malloc.c:3124
#6  0x00007fd2e127b8d8 in __gnu_cxx::new_allocator<std::string>::deallocate(std::string*, unsigned long) ()
   from /usr/local/lib/python3.6/dist-packages/tensorflow_io/image/python/ops/_image_ops.so
#7  0x00007fd2e1279c28 in std::_Vector_base<std::string, std::allocator<std::string> >::_M_deallocate(std::string*, unsigned long) ()
   from /usr/local/lib/python3.6/dist-packages/tensorflow_io/image/python/ops/_image_ops.so
#8  0x00007fd2e12789fd in std::_Vector_base<std::string, std::allocator<std::string> >::~_Vector_base() ()
   from /usr/local/lib/python3.6/dist-packages/tensorflow_io/image/python/ops/_image_ops.so
#9  0x00007fd2e12765e7 in std::vector<std::string, std::allocator<std::string> >::~vector() ()
   from /usr/local/lib/python3.6/dist-packages/tensorflow_io/image/python/ops/_image_ops.so
#10 0x00007fd2e1280d32 in tensorflow::OpDefBuilder::~OpDefBuilder() ()
   from /usr/local/lib/python3.6/dist-packages/tensorflow_io/image/python/ops/_image_ops.so
#11 0x00007fd2e1280f68 in tensorflow::register_op::OpDefBuilderWrapper<true>::~OpDefBuilderWrapper() ()
   from /usr/local/lib/python3.6/dist-packages/tensorflow_io/image/python/ops/_image_ops.so
#12 0x00007fd2e128082a in __static_initialization_and_destruction_0 ()
   from /usr/local/lib/python3.6/dist-packages/tensorflow_io/image/python/ops/_image_ops.so
#13 0x00007fd2e1280a88 in _GLOBAL__sub_I_dataset_ops.cc () from /usr/local/lib/python3.6/dist-packages/tensorflow_io/image/python/ops/_image_ops.so
#14 0x00007fd30285d733 in call_init (env=0x1c71a30, argv=0x7ffe8a9b6188, argc=1, l=<optimized out>) at dl-init.c:72
#15 _dl_init (main_map=main_map@entry=0x201fc00, argc=1, argv=0x7ffe8a9b6188, env=0x1c71a30) at dl-init.c:119
#16 0x00007fd3028621ff in dl_open_worker (a=a@entry=0x7ffe8a9b3340) at dl-open.c:522
#17 0x00007fd3025c32df in __GI__dl_catch_exception (exception=exception@entry=0x7ffe8a9b3320, operate=operate@entry=0x7fd302861dc0 <dl_open_worker>, 
    args=args@entry=0x7ffe8a9b3340) at dl-error-skeleton.c:196
#18 0x00007fd3028617ca in _dl_open (file=0x3a4b840 "/usr/local/lib/python3.6/dist-packages/tensorflow_io/image/python/ops/_image_ops.so", 
    mode=-2147483646, caller_dlopen=0x7fd2e9a1728a <tensorflow::internal::LoadLibrary(char const*, void**)+26>, nsid=<optimized out>, argc=1, 
---Type <return> to continue, or q <return> to quit--- 
```

**Comment by yongtang** [comment]
Likely we may have to rebuild tensorflow-io with TF 1.13 and 2.0 separately and release separate versions for them.

**Comment by yongtang** [comment]
With #106 merged, we have the TF 2.0 supports now. The R still has some issues but it is likely caused by upstream R package (tfdatasets of tensorflow). We could follow up to address those issues later.
